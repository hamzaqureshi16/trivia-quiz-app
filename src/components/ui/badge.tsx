import * as React from 'react';
import { Slot } from '@radix-ui/react-slot';
import { cva, type VariantProps } from 'class-variance-authority';

import { cn } from '@/lib/utils';

const badgeVariants = cva(
  'inline-flex items-center justify-center rounded-md border px-4 py-2 text-sm font-medium transition-all hover:cursor-pointer w-full',
  {
    variants: {
      variant: {
        default: 'border-muted-foreground/30 text-foreground bg-background hover:bg-background/90',
        selected: 'border-primary bg-primary/5 text-primary',
        correct: 'border-green-600 bg-green-600 text-white',
        incorrect: 'border-red-600 bg-red-600 text-white',
        unselected: 'border-green-500 bg-green-500/10 text-green-600',
        secondary:
          'border-transparent bg-secondary text-secondary-foreground [a&]:hover:bg-secondary/90',
        destructive: 'border-transparent bg-destructive text-white [a&]:hover:bg-destructive/90',
        outline:
          'text-foreground hover:bg-black hover:text-white [a&]:hover:bg-accent [a&]:hover:text-accent-foreground',
      },
    },
    defaultVariants: {
      variant: 'default',
    },
  },
);

function Badge({
  className,
  variant,
  asChild = false,
  ...props
}: React.ComponentProps<'span'> & VariantProps<typeof badgeVariants> & { asChild?: boolean }) {
  const Comp = asChild ? Slot : 'span';

  return (
    <Comp data-slot="badge" className={cn(badgeVariants({ variant }), className)} {...props} />
  );
}

export { Badge, badgeVariants };
