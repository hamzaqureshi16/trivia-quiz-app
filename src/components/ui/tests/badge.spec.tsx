import { render, screen } from '@testing-library/react';
import { Badge } from '../badge';
import '@testing-library/jest-dom';

describe('Badge component', () => {
  it('renders with "correct" variant', () => {
    render(<Badge variant="correct">Correct Answer</Badge>);
    const badge = screen.getByText('Correct Answer');
    expect(badge).toBeInTheDocument();
    expect(badge).toHaveClass('bg-green-600');
  });

  it('renders with "incorrect" variant', () => {
    render(<Badge variant="incorrect">Wrong Answer</Badge>);
    const badge = screen.getByText('Wrong Answer');
    expect(badge).toBeInTheDocument();
    expect(badge).toHaveClass('bg-red-600');
  });

  it('renders with "outline" variant by default', () => {
    render(<Badge>Neutral</Badge>);
    const badge = screen.getByText('Neutral');
    expect(badge).toBeInTheDocument();
  });

  it('matches snapshot', () => {
    const { asFragment } = render(<Badge variant="correct">Snapshot</Badge>);
    expect(asFragment()).toMatchSnapshot();
  });
});
