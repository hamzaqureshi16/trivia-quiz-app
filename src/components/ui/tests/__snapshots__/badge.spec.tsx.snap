// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge component matches snapshot 1`] = `
<DocumentFragment>
  <span
    class="inline-flex items-center justify-center rounded-md border px-4 py-2 text-sm font-medium transition-all hover:cursor-pointer w-full border-green-600 bg-green-600 text-white"
    data-slot="badge"
  >
    Snapshot
  </span>
</DocumentFragment>
`;
